src/contrast.rs

```diff
@@ -266,6 +268,85 @@ pub fn equalize_histogram(image: &GrayImage) -> GrayImage {
     out
 }
 
+/// Scales each pixel in the image using the scaling effect to take the input_min and input_max
+/// pair to the output_min and output_max pair.
+///
+/// # Example
+/// (50, 100) -> (0, 255) would make 8 -> 0 since it saturates outside the `u8` range.
+/// 200 -> 255 since it also saturates outside the `u8` range.
+/// 50 -> 0, and 100 -> 255 by definition of the scaling pairs.
+/// All values between 50 and 100 are then linearly interpolated into the output range.
+/// Such as 75 -> 128
+///
+///
+/// # Panic
+/// This function panics if `input_min` > `input_max` or `output_min` > `output_max`.
+pub fn scale_linear(
+    image: &GrayImage,
+    input_min: u8,
+    input_max: u8,
+    output_min: u8,
+    output_max: u8,
+) -> GrayImage {
+    let mut out = image.clone();
+    scale_linear_mut(&mut out, input_min, input_max, output_min, output_max);
+    out
+}
+
+/// Scales each pixel in the image using the scaling effect to take the input_min and input_max
+/// pair to the output_min and output_max pair.
+///
+/// # Example
+/// (50, 100) -> (0, 255) would make 8 -> 0 since it saturates outside the `u8` range.
+/// 200 -> 255 since it also saturates outside the `u8` range.
+/// 50 -> 0, and 100 -> 255 by definition of the scaling pairs.
+/// All values between 50 and 100 are then linearly interpolated into the output range.
+/// Such as 75 -> 128
+///
+///
+/// # Panic
+/// This function panics if `input_min` > `input_max` or `output_min` > `output_max`.
+pub fn scale_linear_mut(
+    image: &mut GrayImage,
+    input_min: u8,
+    input_max: u8,
+    output_min: u8,
+    output_max: u8,
+) {
+    assert!(
+        input_min <= input_max,
+        "input_min must be smaller or equal to input_max"
+    );
+    assert!(
+        output_min <= output_max,
+        "output_min must be smaller or equal to output_max"
+    );
+
+    let input_min: u16 = input_min.into();
+    let input_max: u16 = input_max.into();
+    let output_min: u16 = output_min.into();
+    let output_max: u16 = output_max.into();
+
+    let input_width = input_max - input_min;
+    let output_width = output_max - output_min;
+
+    let f = |_, _, pixel: Luma<u8>| {
```

Comment:
`map_subpixels_mut` would let you write this a bit more concisely - it operates on pixel intensities rather than `(x, y, p)` tuples.